src/common/dao/group/usergroup.go

```diff
@@ -137,3 +160,15 @@ func OnBoardUserGroup(g *models.UserGroup, keyAttribute string, combinedKeyAttri
 
 	return nil
 }
+
+// GetOrCreateHTTPUserGroup - If there is a group with same name, return the group id, or create a new one and return ID
+func GetOrCreateHTTPUserGroup(groupName string) (int, error) {
```

Comment:
1. This does not have to be restricted to HTTP group?
2. Is it safe in concurrent situation?  I think it would be better you call `Add` first and if there's a conflict error you get the ID.

Reply:
1. It can be more general.
2.It is not concurrent safe only when the group name is unique.